warehouse/views.py

```diff
@@ -211,11 +222,17 @@ def search(request):
         first, *_ = cls.classifier.split(' :: ')
         available_filters[first].append(cls.classifier)
 
+    def filter_key(item):
+        try:
+            return SEARCH_FILTER_ORDER.index(item[0])
+        except ValueError:
+            return item[0]
```

Comment:
This is wrong I believe? If passed a classifier that isn't in ``SEARCH_FILTER_ORDER`` it's going to raise an exception because this function is going to return an int if it's in ``SEARCH_FILTER_ORDER`` and a str if it's not.

Perhaps this function should look more like:

```python
def filter_key(item):
    try:
        return (0, SEARCH_FILTER_ORDER.index(item[0], item[0])
    except ValueError:
        return (1, 0, item[0])
```

That should ensure that even if it gets a classifier it doesn't know that (A) it comes last in the list and (B) sorts everything by the defined order, then by alphabetical.